### PR TITLE
Threshold notifier

### PR DESCRIPTION
**Pull request by rem1niscence**
Several refactors for simplifying operations and me doing some typescript magic for scientific purposes, most important part is both logs for apps and load balancers going over thresholds

### CODE REVIEW

**Comment by rem1niscence** [comment]
if you can forgive lodash.get (which I get from a sole npm package, not the whole lodash itself) good things will happen to you

**Review by Evalir — APPROVED** [review]
LGTM! Just some comments

**Comment by Evalir** [comment]
Also, ugh, lodash lol

File: app/src/functions/notify-endpoint-usage/app.ts [hunk 1]
```diff
@@ -184,40 +70,23 @@ exports.handler = async (_: any, context: Context) => {
...
 
-  const networkApps: Map<string, Application> = new Map<string, Application>()
-  networkData.forEach(app => networkApps.set(app.publicKey, app))
+  const networkApps = convertToMap(networkData, 'publicKey')
 
-  let dbApps: IApplication[] = []
-  let loadBalancers: ILoadBalancer[] = []
+  const apps: Map<string, IApplication> = convertToMap(await retryEvery(
+    // @ts-ignore
+    getModelFromDBOrCache.bind(null, redis, ApplicationModel, 'nt-applications')),
```
[the remark this answers is not in the document]
Comment (rem1niscence) [reply app/src/functions/notify-endpoint-usage/app.ts:77]: it is, let's not make this an occurrence (note to self)

**Pull request merged**